### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#1451]
#### For bugs
- Rule Id: SC1009, SC1073, SC1072
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

target_file=$1
compare=$2
compare_file=$3

if [ "$target_file" -nt "$compare_file" ]
then
   echo "$target_file" is "$compare" "$compare_file" 
else
   echo "$target_file" is not "$compare" "$compare_file"
fi
   
if [ "$target_file" "$compare" "$compare_file" ]
then
   echo "$target_file" is "$compare" "$compare_file" 
else
   echo "$target_file" is not "$compare" "$compare_file"
fi

```

#### Here's what shellcheck currently says:
```
Line 14	SC1009: The mentioned syntax error was in this if expression.
Line 14	SC1073: Couldn't parse this test expression. Fix to allow more checks.
Line 14	SC1072: Expected comparison operator (don't wrap commands in []/[[]]). Fix any mentioned problems and try again.
```
#### Here's what I wanted or expected to see:

I want to be able to tell shellcheck that the line 14 isn't a syntax error. This is a contrived example, but the actual code is a shell script that takes --newer and --older and sets $compare as needed, instead of needing an if/else on the compare. Perhaps we could do something with a new directive:

```
#shellcheck disable_until=SC1009
if [ "$target_file" "$compare" "$compare_file" ]
then
   echo "$target_file" is "$compare" "$compare_file" 
else
   echo "$target_file" is not "$compare" "$compare_file"
fi
#shellcheck reenable=SC1009
```

Thoughts?

**Comment by EricLuehrsen** [comment]
Other lint tools have such syntax. Usually they look like command line options as comments with some kind of stack-save. More important for C/C++ libraries and hardware drivers rather than shell, but you can disable in depth, and the enable walks one of those back.
```
#shellcheck -save -eSC1000 -eSC1001
... stuff ...
#shellcheck -restore
```
